pkl-core/src/main/java/org/pkl/core/messaging/Messages.java

```diff
@@ -0,0 +1,520 @@
+      return contents;
+    }
+
+    public @Nullable String getError() {
+      return error;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (!(o instanceof ReadResourceResponse that)) {
+        return false;
+      }
+
+      return evaluatorId == that.evaluatorId
+          && Objects.equals(contents, that.contents)
+          && Objects.equals(error, that.error);
+    }
+
+    @Override
+    public int hashCode() {
+      int result = Long.hashCode(evaluatorId);
+      result = 31 * result + Objects.hashCode(contents);
+      result = 31 * result + Objects.hashCode(error);
+      return result;
+    }
+  }
+
+  public static final class ReadModuleRequest extends Base.Request implements Message.Request {
+
+    private final long evaluatorId;
+    private final URI uri;
+
+    public ReadModuleRequest(long requestId, long evaluatorId, URI uri) {
+      super(Type.READ_MODULE_REQUEST, requestId);
+      this.evaluatorId = evaluatorId;
+      this.uri = uri;
+    }
+
+    public long getEvaluatorId() {
+      return evaluatorId;
+    }
+
+    public URI getUri() {
+      return uri;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (!(o instanceof ReadModuleRequest that)) {
+        return false;
+      }
+
+      return evaluatorId == that.evaluatorId && uri.equals(that.uri);
+    }
+
+    @Override
+    public int hashCode() {
+      int result = Long.hashCode(evaluatorId);
+      result = 31 * result + uri.hashCode();
+      return result;
+    }
+  }
+
+  public static final class ReadModuleResponse extends Base.Response implements Client.Response {
+
+    private final long evaluatorId;
+    private final @Nullable String contents;
+    private final @Nullable String error;
+
+    public ReadModuleResponse(
+        long requestId, long evaluatorId, @Nullable String contents, @Nullable String error) {
+      super(Type.READ_MODULE_RESPONSE, requestId);
+      this.evaluatorId = evaluatorId;
+      this.contents = contents;
+      this.error = error;
+    }
+
+    public long getEvaluatorId() {
+      return evaluatorId;
+    }
+
+    public @Nullable String getContents() {
+      return contents;
+    }
+
+    public @Nullable String getError() {
+      return error;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (!(o instanceof ReadModuleResponse that)) {
+        return false;
+      }
+
+      return evaluatorId == that.evaluatorId
+          && Objects.equals(contents, that.contents)
+          && Objects.equals(error, that.error);
+    }
+
+    @Override
+    public int hashCode() {
+      int result = Long.hashCode(evaluatorId);
+      result = 31 * result + Objects.hashCode(contents);
+      result = 31 * result + Objects.hashCode(error);
+      return result;
+    }
+  }
```

Comment:
I won't block the PR on this, because it's a lot of cruft, but: we should make all of these records instead.

And the message interface getters should have the `get` prefix removed; e.g. `long requestId()` instead of `long getRequestId()` so that the records can conform to the interface easily.

We can leave this for a later refactor, though.

Reply:
I took a look at what this would take and figured it was a small enough effort and a large enough breaking change to be worth doing now!